### PR TITLE
Bump aeson-yaml to 1.0.6.0; add 'hello world' yaml content

### PR DESCRIPTION
**Pull request by patrickmn**
Fixes https://github.com/dhall-lang/dhall-haskell/issues/1516: '2020-02-27' is now single-quoted by both aeson-yaml and HsYAML.

Fixes https://github.com/dhall-lang/dhall-haskell/issues/1617: "hello world" is no longer quoted.

HsYAML was single-quoting when `quoted` was True and double-quoting otherwise. Changed its behavior so it will single-quote strings containing dates and bools in both cases.

### CODE REVIEW

**Comment by patrickmn** [comment]
@sjakobi, I think the "right" long-term solution is to use HsYAML's default (double quoting) both when quoting is True and False, and then change aeson-yaml to always use double quotes. Is there a reason not to do that?

File: dhall-yaml/src/Dhall/Yaml.hs [hunk 1]
```diff
@@ -60,8 +61,17 @@ jsonToYaml json documents quoted =
...
+        | quoted || Text.all isNumberOrDateRelated s || isBoolString =
             Right (YE.untagged, YE.SingleQuoted, s)
+      where
+        isBoolString
+          | Text.length s > 5 = False
+          | otherwise =
+            case Text.toLower s of
+              "true" -> True
+              "false" -> True
+              _ -> False
```
Comment (sjakobi) [inline dhall-yaml/src/Dhall/Yaml.hs]: Could you add tests for this "bool-string" handling?

Comment (patrickmn) [reply dhall-yaml/src/Dhall/Yaml.hs]: done

**Comment by Gabriel439** [comment]
@sjakobi: I was just preserving the existing behavior.  All I know is that single quotes don't interpret escape codes, whereas double quotes do

**Comment by Gabriel439** [comment]
@patrickmn: You can fix the "hydra" build failure by running:

```
$ cabal update
$ cabal2nix cabal://aeson-yaml-1.0.6 > ./nix/aeson-yaml.nix
```

**Comment by patrickmn** [comment]
Ok, I guess let's not fix it if it isn't broken (\*cough\*YAML\*cough\*). We can consider a change if there are any more discrepancies.

**Comment by sjakobi** [comment]
> Ok, I guess let's not fix it if it isn't broken (*cough*YAML*cough*). We can consider a change if there are any more discrepancies.

Are you referring to your plan above for the "right" solution (with more double-quoting IIUC)?

Actually I didn't quite get the motivation that plan. Is it to avoid the complex `style` decision in `jsonToYaml`?

In general I'm all for finding a sweet spot with consistent and as-simple-as-possible-but-no-simpler encoding rules! But I don't have good grasp of the intricacies involved right now.

**Comment by Gabriel439** [comment]
@patrickmn: The "hydra" build is failing on this warning:

```
tasty/Main.hs:17:1: error: [-Wunused-imports, -Werror=unused-imports]
    The qualified import of ‘Test.Tasty.ExpectedFailure’ is redundant
      except perhaps to import instances from ‘Test.Tasty.ExpectedFailure’
    To import instances alone, use: import Test.Tasty.ExpectedFailure()
   |
17 | import qualified Test.Tasty.ExpectedFailure
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by patrickmn** [comment]
> Are you referring to your plan above for the "right" solution (with more double-quoting IIUC)?

Yeah, I mean let's merge this, and if inconsistencies continue to be a headache, we can discuss whether a change like only using double-quoting makes sense.

> Actually I didn't quite get the motivation that plan. Is it to avoid the complex `style` decision in `jsonToYaml`?

And special-casing in aeson-yaml to single-quote instead of double-quote.

> In general I'm all for finding a sweet spot with consistent and as-simple-as-possible-but-no-simpler encoding rules! But I don't have good grasp of the intricacies involved right now.

Same :)

**Comment by sjakobi** [comment]
@patrickmn Would you please allow a bit more time for discussion and wait for an approval from @Gabriel439 or me the next time? (Unless Gabriel gave you permission to merge without approval, which I may have missed.)

In this case I'm a bit concerned about the churn in the YAML output formatting, and was wondering whether we should delay this PR until we have properly considered the double-quoting solution you mentioned. No biggie though, and no need to revert IMHO.

**Comment by patrickmn** [comment]
Sure, my apologies. I thought you had approved this change and was merely asking questions about the long-term suggestion rationale.

**Comment by sjakobi** [comment]
No worries! I'm aware that I sometimes don't communicate very clearly, but I do signal explicit approval via GitHub's "Approve" button.

I guess I've been a bit frustrated about the amount of churn and inconsistencies that the move away from `yaml` entailed. At this point I'm pretty optimistic that the worst is over though! ;)

…and I guess I should remind myself sometimes that it's ok when things are just good _enough_! ;)

**Pull request merged**